cds/modules/flows/api.py

```diff
@@ -192,8 +191,14 @@ def run(self):
         db_flag_modified(self.flow_metadata, "payload")
         db.session.commit()
 
+        # saves the initial conditon of files in payload even after creating version_id when necessary
+        if has_user_uploaded_file is None:
```

Comment:
Thank you for this fix!

Is this line necessary?
Maybe you can change line 179 and put:
`has_user_uploaded_file = True if "version_id" in payload else False`

Actually, I am wondering if we cannot simply change the condition above:
```diff
- if not has_user_uploaded_file and has_remote_file_to_download:
+ if has_remote_file_to_download:
```